### PR TITLE
Implemented 'object_properties' in BaseExtractor

### PR DESCRIPTION
**Pull request by alejoe91**
@samuelgarcia @colehurwitz 

This implements `object_properties`. It comes very handy for example to store channel locations in a sorting extractor.

Let's discuss!

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Could we call it "annotations" ? It is not a joke.

**Comment by alejoe91** [comment]
> Could we call it "annotations" ? It is not a joke.

ok for me

**Comment by alejoe91** [comment]
@samuelgarcia better?

File: spikeextractors/baseextractor.py [hunk 1]
```diff
@@ -271,11 +273,71 @@ def allocate_array(self, memmap, shape=None, dtype=None, name=None, array=None):
                 arr = np.zeros(shape, dtype=dtype)
         return arr
 
+    def set_annotation(self, annotation_name, value, overwrite=False):
```
Comment (samuelgarcia) [inline spikeextractors/baseextractor.py:276]: in neo we call annotate(self, **kargs):
and this support several annoation.

File: spikeextractors/baseextractor.py [hunk 2]
```diff
@@ -271,11 +273,71 @@ def allocate_array(self, memmap, shape=None, dtype=None, name=None, array=None):
...
+            The data associated with the given property name. Could be many
+            formats as specified by the user
+        '''
+        if annotation_name not in self._annotations.keys():
+            print(f"{annotation_name} is not an annotation")
+            return None
+        else:
+            return deepcopy(self._annotations[annotation_name])
+
+    def get_annotation_names(self):
```
Comment (samuelgarcia) [inline spikeextractors/baseextractor.py:317]: get_annotations_keys()

File: spikeextractors/baseextractor.py (hunk 1 above)
Comment (alejoe91) [reply spikeextractors/baseextractor.py:276]: So the thing I don't like about `**kwargs` is that it prevents using spaces..but I can call it annotate

File: spikeextractors/baseextractor.py (hunk 2 above)
Comment (alejoe91) [inline spikeextractors/baseextractor.py:317]:
```suggestion
    def get_annotation_keys(self):
```

File: spikeextractors/baseextractor.py (hunk 1 above)
Comment (samuelgarcia) [reply spikeextractors/baseextractor.py:276]: space for what ? for annotation name ?
You can, annotate({'my spaced key': v}) is this is really needed.

Comment (alejoe91) [reply spikeextractors/baseextractor.py:276]: I would keep the same signature as it is now

**Comment by samuelgarcia** [comment]
Ok lets merge.

File: spikeextractors/baseextractor.py [hunk 3]
```diff
@@ -273,26 +273,26 @@ def allocate_array(self, memmap, shape=None, dtype=None, name=None, array=None):
                 arr = np.zeros(shape, dtype=dtype)
         return arr
 
-    def set_annotation(self, annotation_name, value, overwrite=False):
+    def annotate(self, annotation_key, value, overwrite=False):
```
Comment (samuelgarcia) [inline spikeextractors/baseextractor.py:276]: This is realy neo friendly :)

**Pull request merged**